pkg/cloud/vsphere/services/govmomi/net/net_test.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package net_test
+
+import (
+	"testing"
+
+	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/services/govmomi/net"
+)
+
+func TestIsExternalIPAddr(t *testing.T) {
+	testCases := []struct {
+		name      string
+		ipAddr    string
+		expectErr bool
+	}{
+		{
+			name:      "valid-ipv4",
+			ipAddr:    "192.168.2.33",
+			expectErr: false,
+		},
+		{
+			name:      "valid-ipv6",
+			ipAddr:    "1200:0000:AB00:1234:0000:2552:7777:1313",
+			expectErr: false,
+		},
+		{
+			name:      "localhost",
+			ipAddr:    "127.0.0.1",
+			expectErr: true,
+		},
+		{
+			name:      "link-local-unicast-ipv4",
+			ipAddr:    "169.254.2.3",
+			expectErr: true,
+		},
+		{
+			name:      "link-local-unicast-ipv6",
+			ipAddr:    "fe80::250:56ff:feb0:345d",
+			expectErr: true,
+		},
+		{
+			name:      "link-local-multicast-ipv4",
+			ipAddr:    "224.0.0.252",
+			expectErr: true,
+		},
+		{
+			name:      "link-local-multicast-ipv6",
+			ipAddr:    "FF02:0:0:0:0:0:1:3",
+			expectErr: true,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			if err := net.IsExternalIPAddr(tc.ipAddr); err != nil {
+				t.Log(err)
```

Comment:
can remove this since there's `t.Fatal` below

Reply:
Hi @andrewsykim,

Not when I tried it. It showed both.

Reply:
![image](https://user-images.githubusercontent.com/101085/60135516-82006380-9767-11e9-9111-b7cd61e3cfbe.png)

Reply:
FWIW, I thought so too, but it's different in the test case pattern.

Reply:
Hi @andrewsykim,

I'd like to resolve this. Want to verify I'm not crazy here?

Reply:
Yeah it'll print both.  Looks like @akutz just wants to always Log() the error and only Fatal() when the test expects an error.